modules/services/repositories/src/main/java/au/com/shiftyjelly/pocketcasts/repositories/playback/SimplePlayer.kt

```diff
@@ -265,13 +282,32 @@ class SimplePlayer(val settings: Settings, val statsManager: StatsManager, val c
             }
         } ?: return
 
+        if (FeatureFlag.isEnabled(Feature.CACHE_PLAYING_EPISODE) && location is EpisodeLocation.Stream) {
+            databaseProvider ?: run { databaseProvider = StandaloneDatabaseProvider(context) }
+
+            simpleCache ?: run {
+                simpleCache = SimpleCache(
+                    File(context.cacheDir, "podcasts-cache"),
+                    LeastRecentlyUsedCacheEvictor(MAX_DEVICE_CACHE_SIZE_BYTES.toLong()),
+                    databaseProvider!!,
+                )
+            }
+        }
+
+        val cacheDataSourceFactory: CacheDataSource.Factory? =
+            simpleCache?.let {
+                CacheDataSource.Factory()
+                    .setCache(it)
+                    .setUpstreamDataSourceFactory(httpDataSourceFactory)
+            }
+
         val mediaItem = MediaItem.fromUri(uri)
         val source = if (isHLS) {
-            HlsMediaSource.Factory(dataSourceFactory).createMediaSource(mediaItem)
+            HlsMediaSource.Factory(cacheDataSourceFactory ?: dataSourceFactory)
         } else {
-            ProgressiveMediaSource.Factory(dataSourceFactory, extractorsFactory)
-                .createMediaSource(mediaItem)
-        }
+            ProgressiveMediaSource.Factory(cacheDataSourceFactory ?: dataSourceFactory, extractorsFactory)
```

Comment:
The `cacheDataSourceFactory ?: dataSourceFactory` can be extracted to something like

```kotlin
val sourceFactory = simpleCache?.let {
     CacheDataSource.Factory()
        .setCache(it)
        .setUpstreamDataSourceFactory(httpDataSourceFactory)
} ?: dataSourceFactory
```

Reply:
Yeah. I liked it! Here: [464ddb5](https://github.com/Automattic/pocket-casts-android/pull/1910/commits/464ddb5eecde733ea45057708b2c3cfe18c72cd0)